ros/src/tl_detector/light_classification/tl_classifier_cnn.py

```diff
@@ -0,0 +1,583 @@
+                tf.reduce_sum(self._labels_float * tf.log(self._prediction_softmax),
+                              reduction_indices=[1]),
+                name="cross_entropy")
+            self._loss = tf.reduce_mean(cross_entropy, name='loss')
+            tf.summary.scalar('loss', self._loss)
+
+    def _create_accuracy(self):
+        #pylint: disable=not-context-manager
+        """ define accuracy metric """
+        with tf.name_scope("accuracy"):
+            true_class = tf.argmax(self._labels_float, 1)
+            correct_prediction = tf.equal(true_class, self._prediction_class)
+            self._accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
+            tf.summary.scalar('accuracy', self._accuracy)
+
+    def _create_optimizer(self):
+        #pylint: disable=not-context-manager
+        """ define optimizer to use in training """
+        with tf.name_scope("optimizer"):
+            self._global_step = tf.Variable(0, dtype=tf.int32, trainable=False, name='global_step')
+            if self._learning_rate is not None:
+                self._optimizer = tf.train.AdamOptimizer(self._learning_rate) \
+                    .minimize(self._loss, global_step=self._global_step)
+            else:
+                self._optimizer = tf.train.AdamOptimizer().minimize(self._loss,
+                                                                    global_step=self._global_step)
+
+    def _create_session(self, gpu_mem_fraction=0.9):
+        """ create and configure tensorflow session """
+        # GPU config
+        config = tf.ConfigProto(log_device_placement=False)
+        config.gpu_options.allow_growth = True
+        config.gpu_options.per_process_gpu_memory_fraction = gpu_mem_fraction
+        # session
+        self._session = tf.Session(config=config)
+
+    def __init__(self, learning_rate=None):
+        """ Create calculation graph and session """
+        image_shape = (32, 32, 3)
+        # label converter
+        self._label_converter = TLLabelConverter()
+        # reset graph
+        tf.reset_default_graph()
+        self._tag = 'tl_classifier'
+        # inputs
+        self._features_shape = (None,) + image_shape
+        self._labels_shape = (None, len(self._label_converter.labels()))
+        # model
+        self._create_inputs()
+        self._create_input_transforms()
+        self._conv1 = self._create_layer_1()
+        self._conv2 = self._create_layer_2()
+        self._keep_prob = tf.placeholder(tf.float32, name='dropout_keep_probability')
+        self._logits = self._create_layer_3()
+        self._create_predictions()
+        self._create_loss()
+        self._create_accuracy()
+        self._learning_rate = learning_rate
+        self._create_optimizer()
+        # auxiliary objects
+        self._summaries = tf.summary.merge_all()
+        # session
+        self._create_session(0.9)  # gpu mem fraction to use
+        self._session.run(tf.global_variables_initializer())
```

Comment:
This won't work with a shared session and you don't need that if you just load the model.